includes/woopay/class-woopay-store-api-session-handler.php

```diff
@@ -74,6 +74,15 @@ protected function init_session_from_token() {
 		$this->_data              = (array) $this->get_session( $this->_customer_id, [] );
 	}
 
+	/**
+	 * Return true if the current user has an active session,.
+	 *
+	 * @return bool
+	 */
+	public function has_session() {
```

Comment:
I'm also working on Gift Cards extension compatibility on WooPay, currently when this extension is installed on the merchant site it breaks WooPay because it looks for the `has_session` method on the `WC()->session` that's not implemented on the WooPay session handler, I've preferred implement it on this PR (the first one of the project) to stop breaking WooPay in case this PR is merged earlier.

<img width="901" alt="Screenshot 2023-07-14 at 12 23 19" src="https://github.com/Automattic/woocommerce-payments/assets/1693223/91906460-9b26-459c-b2be-376170c8b875">

<img width="869" alt="Screenshot 2023-07-14 at 12 23 41" src="https://github.com/Automattic/woocommerce-payments/assets/1693223/680750da-556d-46b4-83da-bbf28be046b5">

Reply:
Actually it's breaking the production site too, if the merchant store has the Gift Cards installed (and possibly other extensions that check for the `has_session` method), WooPay doesn't work because it's a method expected to be implemented in the WC session handler, should I continue and remove it to implement only on the Gift Card compatibility PR?